operate/bridge/providers/native_bridge_provider.py

```diff
@@ -0,0 +1,650 @@
+            message = MESSAGE_QUOTE_ZERO
+
+        quote_data = QuoteData(
+            bridge_eta=bridge_eta,
+            elapsed_time=0,
+            message=message,
+            provider_data=None,
+            timestamp=int(time.time()),
+        )
+        bridge_request.quote_data = quote_data
+        bridge_request.status = BridgeRequestStatus.QUOTE_DONE
+
+    def _get_approve_tx(self, bridge_request: BridgeRequest) -> t.Optional[t.Dict]:
+        """Get the approve transaction."""
+        self.logger.info(
+            f"[NATIVE BRIDGE] Get appprove transaction for bridge request {bridge_request.id}."
+        )
+
+        if bridge_request.params["to"]["amount"] == 0:
+            return None
+
+        quote_data = bridge_request.quote_data
+        if not quote_data:
+            return None
+
+        from_address = bridge_request.params["from"]["address"]
+        from_token = bridge_request.params["from"]["token"]
+        to_amount = bridge_request.params["to"]["amount"]
+        from_bridge = self.bridge_contract_adaptor.from_bridge
+        from_ledger_api = self._from_ledger_api(bridge_request)
+
+        if from_token == ZERO_ADDRESS:
+            return None
+
+        approve_tx = registry_contracts.erc20.get_approve_tx(
+            ledger_api=from_ledger_api,
+            contract_address=from_token,
+            spender=from_bridge,
+            sender=from_address,
+            amount=to_amount,
+        )
+        approve_tx["gas"] = 200_000  # TODO backport to ERC20 contract as default
+        BridgeProvider._update_with_gas_pricing(approve_tx, from_ledger_api)
+        BridgeProvider._update_with_gas_estimate(approve_tx, from_ledger_api)
+        approve_tx["gas"] = ceil(approve_tx["gas"] * GAS_ESTIMATE_BUFFER)
+        return approve_tx
+
+    def _get_bridge_tx(self, bridge_request: BridgeRequest) -> t.Optional[t.Dict]:
+        """Get the bridge transaction."""
+        self.logger.info(
+            f"[NATIVE BRIDGE] Get bridge transaction for bridge request {bridge_request.id}."
+        )
+
+        if bridge_request.params["to"]["amount"] == 0:
+            return None
+
+        quote_data = bridge_request.quote_data
+        if not quote_data:
+            return None
+
+        from_ledger_api = self._from_ledger_api(bridge_request)
+        bridge_tx = self.bridge_contract_adaptor.build_bridge_tx(
+            from_ledger_api=from_ledger_api, bridge_request=bridge_request
+        )
+
+        BridgeProvider._update_with_gas_pricing(bridge_tx, from_ledger_api)
+        BridgeProvider._update_with_gas_estimate(bridge_tx, from_ledger_api)
+        bridge_tx["gas"] = ceil(bridge_tx["gas"] * GAS_ESTIMATE_BUFFER)
+        return bridge_tx
+
+    def _update_execution_status(self, bridge_request: BridgeRequest) -> None:
+        """Update the execution status. Returns `True` if the status changed."""
+
+        if bridge_request.status not in (BridgeRequestStatus.EXECUTION_PENDING,):
```

Comment:
looks cleaner
```suggestion
        if bridge_request.status != BridgeRequestStatus.EXECUTION_PENDING:
```

Reply:
It's done for consistency with the remaining methods, where a status should not be in a set of statuses. This is a degenerate case with only 1 status. Will fix.